src/NuGet.Core/NuGet.Packaging/Signing/Utility/SigningUtility.cs

```diff
@@ -272,7 +272,7 @@ public static async Task SignAsync(SigningOptions options, SignPackageRequest si
             {
                 if (unsignedPackageStream != null && !ReferenceEquals(unsignedPackageStream, options.InputPackageStream))
                 {
-                    unsignedPackageStream.Dispose();
+                    await unsignedPackageStream.DisposeAsync();
```

Comment:
According to [the docs for `Stream.DisposeAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.disposeasync?view=net-6.0), this API was added in .NET Core 3.0, which means it won't compile on .NET Framework.